core/src/test/java/com/backblaze/b2/client/B2LargeFileStorerTest.java

```diff
@@ -106,6 +105,49 @@ private B2LargeFileStorer createFromLocalContent() throws B2Exception {
 
     }
 
+    private B2ContentSource createContentSourceWithSize(long size) throws IOException {
+        final B2ContentSource contentSource = mock(B2ContentSource.class);
+        when(contentSource.getContentLength()).thenReturn(size);
+        return contentSource;
+    }
+
+    private B2LargeFileStorer createLargeFileStorerForStartByteTests() throws IOException {
+        final List<B2PartStorer> partStorers = Arrays.asList(
+                new B2UploadingPartStorer(1, createContentSourceWithSize(100)),
+                new B2AlreadyStoredPartStorer(part2),
+                new B2CopyingPartStorer(3, fileId(3)),
+                new B2UploadingPartStorer(1, createContentSourceWithSize(900)));
```

Comment:
duplicate part number?  is that intentional?  see questions above.  :)

Reply:
Good catch! This is now being addressed with the validation logic I added :)